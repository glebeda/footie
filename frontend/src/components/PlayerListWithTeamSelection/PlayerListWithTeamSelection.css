.player-list-container {
    margin-top: 0.6rem;
    overflow-y: auto;
    scroll-behavior: smooth;
  }
  
  .radio-group {
    display: flex;
    justify-content: flex-end;
  }
  