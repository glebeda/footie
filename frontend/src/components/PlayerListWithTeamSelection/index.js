import { Table, TableBody, TableCell, TableContainer, TableHead, TableRow, Paper } from '@mui/material';
import './PlayerListWithTeamSelection.css';
import PlayerRowWithTeamSelection from '../PlayerRowWithTeamSelection'

function PlayerListWithTeamSelection({ players, teamAssignments, handleTeamChange }) {
  
  const mainPlayers = players.filter(player => player.role === 'MAIN');  

  return (
    <TableContainer className='player-list-container' component={Paper}>
      <Table aria-label='simple table'>
        <TableHead>
          <TableRow>
            <TableCell>#</TableCell>
            <TableCell>Name</TableCell>
            <TableCell align='left'>Team</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {mainPlayers.map((player, index) => (
            <PlayerRowWithTeamSelection
              key={`player-${player.playerId}`}
              player={player}
              index={index + 1}
              team={teamAssignments[player.playerId] || ''} 
              handleTeamChange={handleTeamChange}
            />
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}

export default PlayerListWithTeamSelection;
