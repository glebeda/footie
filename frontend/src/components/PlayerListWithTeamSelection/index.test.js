import React from 'react';
import { render, screen } from '@testing-library/react';
import PlayerListWithTeamSelection from './index';

// Mock PlayerRowWithTeamSelection to isolate tests
jest.mock('../PlayerRowWithTeamSelection', () => ({ player, index, team, handleTeamChange }) => (
  <tr>
    <td>{index}</td>
    <td>{player.name}</td>
    <td>{team}</td>
  </tr>
));

describe('PlayerListWithTeamSelection', () => {
  const players = [
    { playerId: '1', name: 'Player One', role: 'MAIN' },
    { playerId: '2', name: 'Player Two', role: 'MAIN' },
    { playerId: '3', name: 'Player Three', role: 'SUBSTITUTE' }
  ];
  const teamAssignments = { '1': 'LIGHTS', '2': 'DARKS' };
  const handleTeamChange = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders the table headers correctly', () => {
    render(
      <PlayerListWithTeamSelection
        players={players}
        teamAssignments={teamAssignments}
        handleTeamChange={handleTeamChange}
      />
    );

    expect(screen.getByText('#')).toBeInTheDocument();
    expect(screen.getByText('Name')).toBeInTheDocument();
    expect(screen.getByText('Team')).toBeInTheDocument();
  });

  it('renders PlayerRowWithTeamSelection components for main players', () => {
    render(
      <PlayerListWithTeamSelection
        players={players}
        teamAssignments={teamAssignments}
        handleTeamChange={handleTeamChange}
      />
    );

    expect(screen.getByText('Player One')).toBeInTheDocument();
    expect(screen.getByText('Player Two')).toBeInTheDocument();
    expect(screen.queryByText('Player Three')).not.toBeInTheDocument();
  });

  it('passes the correct props to PlayerRowWithTeamSelection', () => {
    render(
      <PlayerListWithTeamSelection
        players={players}
        teamAssignments={teamAssignments}
        handleTeamChange={handleTeamChange}
      />
    );

    const firstPlayerRow = screen.getByText('Player One').closest('tr');
    const secondPlayerRow = screen.getByText('Player Two').closest('tr');

    expect(firstPlayerRow).toHaveTextContent('1'); // Index
    expect(firstPlayerRow).toHaveTextContent('Player One');
    expect(firstPlayerRow).toHaveTextContent('LIGHTS');

    expect(secondPlayerRow).toHaveTextContent('2'); // Index
    expect(secondPlayerRow).toHaveTextContent('Player Two');
    expect(secondPlayerRow).toHaveTextContent('DARKS');
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(
      <PlayerListWithTeamSelection
        players={players}
        teamAssignments={teamAssignments}
        handleTeamChange={handleTeamChange}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
