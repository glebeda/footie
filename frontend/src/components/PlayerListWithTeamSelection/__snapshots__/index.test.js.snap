// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerListWithTeamSelection matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root player-list-container css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      aria-label="simple table"
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            #
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Name
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
            scope="col"
          >
            Team
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr>
          <td>
            1
          </td>
          <td>
            Player One
          </td>
          <td>
            LIGHTS
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td>
            Player Two
          </td>
          <td>
            DARKS
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
